x-pack/plugins/reporting/export_types/printable_pdf/server/lib/layouts/preserve_layout.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import path from 'path';
+import {
+  kbn_server,
+  ViewWidthHeight,
+  ViewZoomWidthHeight,
+} from '../../../../../../../../src/server/index';
+import { Layout } from './layout';
+
+interface Pagesizeparams {
+  pageMarginTop: number;
+  pageMarginBottom: number;
+  pageMarginWidth: number;
+  tableBorderWidth: number;
+  headingHeight: number;
+  subheadingHeight: number;
+}
+
+export class Preservelayout extends Layout {
+  public groupCount: number = 1;
+
+  public selectors = {
+    screenshot: '[data-shared-items-container]',
+    renderComplete: '[data-shared-item]',
+    itemsCountAttribute: 'data-shared-items-count',
+    timefilterFromAttribute: 'data-shared-timefilter-from',
+    timefilterToAttribute: 'data-shared-timefilter-to',
+  };
+
+  public height: number = 0;
+  public width: number = 0;
+  public zoom: number = 0;
+
+  constructor(server: kbn_server, id: string, height: number, width: number, zoom: number) {
```

Comment:
I would pass in height and width as an object that maps to size, and get rid of `zoom`.  It was never used in the old code, and I don't think there are any plans to make it configurable (https://github.com/elastic/kibana/issues/20091)

`constructor(server: kbn_server, id: string, size: Size) {`